runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -758,8 +768,19 @@ private void pruneRemovedRouters(@Nonnull Layout layout) {
      */
     private CompletableFuture<Layout> fetchLayout(List<String> layoutServers) {
 
+        // Layout fetch can result in an exception if the systemDownHandler was invoked and the
+        // completable future was completed exceptionally.
+        Layout lastKnownLayout;
```

Comment:
there is a commonality in this code change and the code change above. Can we make one common private method.

Reply:
Added a latestLayout to keep a record of the latest seen layout.